### PR TITLE
710-yc-Diameter

### PR DESCRIPTION
**Pull request by maverickyuwei**
Initiation.

### CODE REVIEW

**Comment by maverickyuwei** [comment]
- [x] Diameter
- [ ] test_Diameter

File: braph2/graph/measures/Diameter.m [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            % DIAMETER = CALCULATE(M) returns the value of the diameter of a
+            % graph.
+            %   
+            % See also Measure, Eccentricity, Radius, GraphBU, GraphWU, MultiplexGraphBU, MultiplexGraphWU.
+            
+            g = m.getGraph();  % graph from measure class
+            L = g.layernumber();
+            
+            if g.is_measure_calculated('Eccentricity')
+                eccentricity = g.getMeasure('Eccentricity').getValue();
```
Comment (annacanal) [inline braph2/graph/measures/Diameter.m:53]: Instead: eccentricity = g.getMeasureValue('Eccentricity');

File: braph2/graph/measures/test_Diameter.m [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+%% Test 1: GraphBU
+A = [
+    0     .1  .2  .25  0;
+    .125  0   0   0    0;
+    .2    .5  0   .25  0;
+    .125  10  0   0    0;
+    0     0   0   0    0
+    ];
+    
+known_diameter_subgraphs = {[1]};
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:12]: known_diameter_subgraphs = {1};

File: braph2/graph/measures/test_Diameter.m [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+g = GraphBU(A);
+diameter = Diameter(g, 'EccentricityRule', 'subgraphs');
+
+assert(isequal(diameter.getValue(), known_diameter_subgraphs), ...
+    [BRAPH2.STR ':Diameter:' BRAPH2.BUG_ERR], ...
+    'Diameter is not being calculated correctly for GraphBU.')
+
+known_diameter_default = {[Inf]'};
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:21]: known_diameter_default = {Inf};

File: braph2/graph/measures/test_Diameter.m [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+%% Test 2: GraphWU
+A = [
+    0     .1  .2  .25  0;
+    .125  0   0   0    0;
+    .2    .5  0   .25  0;
+    .125  10  0   0    0;
+    0     0   0   0    0
+    ];
+    
+known_diameter_subgraphs = {[5]};
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:38]: known_diameter_subgraphs = {5};

File: braph2/graph/measures/test_Diameter.m [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+g = GraphWU(A);
+diameter = Diameter(g, 'EccentricityRule', 'subgraphs');
+
+assert(isequal(diameter.getValue(), known_diameter_subgraphs), ...
+    [BRAPH2.STR ':Diameter:' BRAPH2.BUG_ERR], ...
+    'Diameter is not being calculated correctly for GraphWU.')
+
+known_diameter_default = {[Inf]};
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:47]: known_diameter_default = {Inf};

File: braph2/graph/measures/test_Diameter.m [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+A = {
+    A11     A12
+    A21     A22
+    };
+
+known_diameter_subgraphs = {
+                                [1]
+                                [1]
+                                };
+
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:82]: known_diameter_subgraphs = {
                                1
                                1
                                };

File: braph2/graph/measures/test_Diameter.m [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+assert(isequal(diameter.getValue(), known_diameter_subgraphs), ...
+    [BRAPH2.STR ':Diameter:' BRAPH2.BUG_ERR], ...
+    'Diameter is not being calculated correctly for MultiplexGraphBU.')
+
+known_diameter_default = {
+                          [Inf]
+                          [Inf]
+                         };
+
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:94]: known_diameter_default = {
                          Inf
                          Inf
                         };

File: braph2/graph/measures/test_Diameter.m [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+A = {
+    A11     A12
+    A21     A22
+    };
+
+known_diameter_subgraphs = {
+                            [5]
+                            [5]
+                           };
+
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:128]: known_diameter_subgraphs = {
                            5
                            5
                           };

File: braph2/graph/measures/test_Diameter.m [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+diameter = Diameter(g, 'EccentricityRule', 'subgraphs');
+
+assert(isequal(diameter.getValue(), known_diameter_subgraphs), ...
+    [BRAPH2.STR ':Diameter:' BRAPH2.BUG_ERR], ...
+    'Diameter is not being calculated correctly for MultiplexGraphWU.')
+
+known_diameter_default = {
+                          [Inf]
+                          [Inf]
+                         };
```
Comment (annacanal) [inline braph2/graph/measures/test_Diameter.m:139]: known_diameter_default = {
                          Inf
                          Inf
                         };

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2/graph/measures/Diameter.m [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    %   getName                     - returns the name of diameter measure.
+    %   getDescription              - returns the description of diameter measure.
+    %   getAvailableSettings        - returns the settings available to the class.
+    %   getMeasureFormat            - returns the measure format
+    %   getMeasureScope             - returns the measure scope
+    %   getParametricity            - returns the parametricity of the measure  
+    %   getMeasure                  - returns the diameter class.
+    %   getCompatibleGraphList      - returns a list of compatible graphs.
+    %   getCompatibleGraphNumber    - returns the number of compatible graphs.
+    %
```
[the remark this answers is not in the document]
Comment (maverickyuwei) [reply braph2/graph/measures/Diameter.m]: 7c6b5ba

File: braph2/graph/measures/Diameter.m (hunk 1 above)
Comment (maverickyuwei) [reply braph2/graph/measures/Diameter.m]: 259455e

File: braph2/graph/measures/test_Diameter.m (hunk 2 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 3 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 4 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 5 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 6 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 7 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 8 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

File: braph2/graph/measures/test_Diameter.m (hunk 9 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_Diameter.m]: 6dfdc21

**Review by annacanal — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**